x/ibc/light-clients/06-solomachine/spec/01_concepts.md

```diff
@@ -71,3 +107,47 @@ timestampedSignatureData := &types.TimestampedSignatureData{
 
 proof, err := cdc.MarshalBinaryBare(timestampedSignatureData)
 ```
+
+## Updates By Header
+
+An update by a header will only succeed if:
+- the header provided is parseable to solo machine header
+- the header sequence matches the current sequence
+- the header timestamp is greater than or equal to the consensus state timestamp
+- the currently registered public key generated the proof
+
+If the update is successful:
+- the public key is updated
+- the diversifier is updated
+- the timestamp is updated
+- the sequence is incremented by 1
+- the new consensus state is set in the client state 
+
+## Updates By Proposal
+
+An update by a governance proposal will only succeed if:
+- the header provided is parseable to solo machine header
+- the `AllowUpdateAfterProposal` client parameter is set to `true`
+- the new header public key does not equal the consensus state public key
+
+If the update is successful:
+- the public key is updated
+- the diversifier is updated
+- the timestamp is updated
+- the sequence is updated
+- the new consensus state is set in the client state 
+- the client is unfrozen (if it was previously frozen)
+
+## Misbehaviour
+
+Misbehaviour handling will only succeed if:
+- the misbehaviour provided is parseable to solo machine misbehaviour
+- the client is not already frozen
+- the current public key signed over two unique data messages at the same sequence and diversifier. 
+
+If the misbehaviour is successfully processed:
+- the client is frozen by setting the frozen sequence to the misbehaviour sequence
+
+## Upgrades
+
+Upgrades to solo machine light clients are not supported. 
```

Comment:
It might be useful to write a one-liner about the reason why they are not supported